backend/main.py

```
@@ -11,208 +17,180 @@
             await asyncio.sleep(0.1)
-            
+
             contents = await file.read()
-            image = Image.open(io.BytesIO(contents))
-            
-            if image.mode != 'RGB':
-                image = image.convert('RGB')
-                
+            image = Image.open(io.BytesIO(contents)).convert('RGB')
+
             current_dir = os.path.dirname(os.path.abspath(__file__))
```

Ruff (0.15.13)

[warning] Async functions should not use os.path methods, use trio.Path or anyio.path

(ASYNC240)